### PR TITLE
Updating prompt to accept max email and date range

### PR DESCRIPTION
**Pull request by cplummer-uwm**
Added message filtering by date

### CODE REVIEW

File: outlook_analyzer.py [hunk 1]
```diff
@@ -277,13 +319,45 @@ def main():
...
     if user_input == "Y" or user_input == "y":
-        # TO DO: Add more input questions for time range and max number of emails to pull
 
-        extract_outlook_information()
+        # Check if user provided an actual integer
+        while True:
+            max_email_number_to_extract_input = input("Max number of email messages you would like to extract (hit Enter for default: 500)?") or 500
+
+            try:
+                int(max_email_number_to_extract_input)
```
Comment (lisashub) [inline outlook_analyzer.py:333]: Would you be willing to also add a check that the input is at least 50? Word cloud feature currently assumes at least 50 emails to represent. Probably would need to adjust the prompt, as well, to reflect this requirement.

File: outlook_analyzer.py [hunk 2]
```diff
@@ -277,13 +319,45 @@ def main():
...
+                break;
+            except ValueError:
+                print("This is not an integer. Please enter a valid integer")
+
+        # Check if user entered in proper format for day/month
+        while True:
+            date_start_input = input("Starting time to pull email from in months or days ago (e.g. 10m, 12d) (Hit enter for default: 12 months ago)?") or "12m"
```
Comment (lisashub) [inline outlook_analyzer.py]: Prompt wording feels little stiff. Would you be willing to change it something like "From how far back would you like to collect and analyze emails in months or days (e.g. 10m, 12d)?

File: outlook_analyzer.py [hunk 3]
```diff
@@ -277,13 +319,45 @@ def main():
...
+            try:
+                int(date_start_input[0:-1])
+                if date_start_input[-1] == "m" or date_start_input[-1] == "d":
+                    break;
+            except ValueError:
+                print("This is not a valid format. Please enter as '##m' or '##d' where d is for days and m is for months  (e.g. 10d or 1m")
+
+        # Check if user entered in proper format for day/month
+        while True:
+            date_end_input = input("Ending time to pull email from in number of months or days ago (e.g. 1m, 10d) (Hit enter for default: today)?") or "0m"
```
Comment (lisashub) [inline outlook_analyzer.py]: Similar to start time - maybe try something like "What's the cutoff for the most recent emails you'd like to collect and analyze in months or days (e.g. 1m, 10d)?" ...or something. Finding this one a bit harder to make user-friendly, haha.

File: outlook_analyzer.py [hunk 4]
```diff
@@ -96,10 +137,11 @@ def extract_outlook_information(): #to modify as new features required
             append_to_error_list(str(sys._getframe().f_code.co_name),str(e))
         
         message_counter_int = message_counter_int + 1
-        
-        if message_counter_int == 500:
+
+        # Check if max number of email has been reached       
+        if message_counter_int >= int(max_email_number_to_extract_input):
```
Comment (lisashub) [inline outlook_analyzer.py:142]: Out of pure curiosity, what was your reasoning behind casting the max email variable as an int here? Tested without the cast and it ran fine, but thinking maybe you  had your reasons.

**Review by lisashub — CHANGES_REQUESTED** [review]
Looking super slick! Biggest change request would be ensuring that users extract at least 50 emails to support word cloud (which is getting fixed up right now). There's also some input prompt change suggestions that I think will translate better for users.

Also I finally found out how to do the fancy comments tied to script lines... I think. We'll see how these show up after I finish this review! :D

File: outlook_analyzer.py (hunk 4 above)
Comment (cplummer-uwm) [reply outlook_analyzer.py:142]: Two reasons:
1. You cannot compare an int value with this operator ">=" against a string. The counter variable is an int and the variable max_email_number_to_extract_input is a string provided by the user. You get this error:
```Traceback (most recent call last):
  File "c:\Users\churn\Documents\GitHub\outlookanalyzer\outlook_analyzer.py", line 373, in <module>
    main()
  File "c:\Users\churn\Documents\GitHub\outlookanalyzer\outlook_analyzer.py", line 360, in main
    extract_outlook_information(max_email_number_to_extract_input,date_start_input,date_end_input)
  File "c:\Users\churn\Documents\GitHub\outlookanalyzer\outlook_analyzer.py", line 142, in extract_outlook_information
    if message_counter_int >= max_email_number_to_extract_input:
TypeError: '>=' not supported between instances of 'int' and 'str'
```
2. If you use the default value by pressing enter, it gets passed in as an integer, and not a string and works but if you enter anything else, it needs to be converted to an int.

File: outlook_analyzer.py (hunk 2 above)
Comment (cplummer-uwm) [reply outlook_analyzer.py]: Your wording is better, updated to that. Thanks for the more clear wording. :)

File: outlook_analyzer.py (hunk 3 above)
Comment (cplummer-uwm) [reply outlook_analyzer.py]: I like your words better, changed to that. Thanks.  :-)

File: outlook_analyzer.py (hunk 1 above)
Comment (cplummer-uwm) [reply outlook_analyzer.py:333]: Added the min value to 50 and a max of 100000.

**Comment by cplummer-uwm** [comment]
@lisashub Updated the script, please review again.

**Review by lisashub — APPROVED** [review]
Looks good - many thanks!

**Pull request merged**